packages/core/src/2d/assembler/TiledSpriteAssembler.ts

```diff
@@ -18,7 +19,7 @@ export class TiledSpriteAssembler {
   static _uvRow = new DisorderedArray<number>();
   static _uvColumn = new DisorderedArray<number>();
 
-  static resetData(renderer: SpriteRenderer, vertexCount: number): void {
+  static resetData(renderer: SpriteRenderer | UIImage, vertexCount: number): void {
```

Comment:
Same as commented above.
we have renderer: SpriteRenderer | UIImage, which includes only two classes.
If we anticipate this pattern expanding or being reused in the future, it might still be beneficial to define a type like Renderable for consistency and maintainability, even though there are currently only two classes. However, if this combination is unique to this specific case, leaving it as-is might be more straightforward.